reactive/flannel.py

```diff
@@ -245,6 +248,13 @@ def cleanup_deployment():
 
 def get_flannel_subnet():
     ''' Returns the flannel subnet reserved for this unit '''
-    with open('/run/flannel/subnet.env') as f:
-        raw_data = dict(line.strip().split('=') for line in f)
-    return raw_data['FLANNEL_SUBNET']
+    try:
+        with open('/run/flannel/subnet.env') as f:
+            raw_data = dict(line.strip().split('=') for line in f)
+        return raw_data['FLANNEL_SUBNET']
+    except FileNotFoundError as e:
+        raise FlannelSubnetNotFound() from e
```

Comment:
yeah, i gave it more thought and came to that conclusion myself. It didn't seem intuitive at first, but i'd rather carry 2 lines of code for clarity than be more pythonic and hate myself 6 months from now.